### PR TITLE
[#5173] feat(hadoop-catlog): Support OSS fileset for Gravitino.

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

Add OSS fileset support. 

### Why are the changes needed?

OSS is widely used in China and Asia. 

Fix: #5173 

### Does this PR introduce _any_ user-facing change?

N/A.

### How was this patch tested?

Test Locally.

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -154,12 +156,14 @@ hadoop3-common = { group = "org.apache.hadoop", name = "hadoop-common", version.
...
 hadoop3-minicluster = { group = "org.apache.hadoop", name = "hadoop-minicluster", version.ref = "hadoop-minikdc"}
 hadoop3-gcs = { group = "com.google.cloud.bigdataoss", name = "gcs-connector", version.ref = "hadoop3-gcs"}
+hadoop3-oss = { group = "org.apache.hadoop", name = "hadoop-aliyun", version.ref = "hadoop3-aliyun"}
 htrace-core4 = { group = "org.apache.htrace", name = "htrace-core4", version.ref = "htrace-core4" }
 airlift-json = { group = "io.airlift", name = "json", version.ref = "airlift-json"}
 airlift-resolver = { group = "io.airlift.resolver", name = "resolver", version.ref = "airlift-resolver"}
 httpclient5 = { group = "org.apache.httpcomponents.client5", name = "httpclient5", version.ref = "httpclient5" }
 mockserver-netty = { group = "org.mock-server", name = "mockserver-netty", version.ref = "mockserver" }
 mockserver-client-java = { group = "org.mock-server", name = "mockserver-client-java", version.ref = "mockserver" }
+commons-lang = { group = "commons-lang", name = "commons-lang", version.ref = "commons-lang" }
```
Comment (jerryshao) [inline gradle/libs.versions.toml:166]: You need to update the LICENSE.bin for your new added dependency.

File: bundles/aliyun-bundle/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  compileOnly(libs.hadoop3.common)
+  implementation(libs.hadoop3.oss)
+
+  // oss needs StringUtils from commons-lang or the following error will occur in 3.1.0
+  // java.lang.NoClassDefFoundError: org/apache/commons/lang/StringUtils
+  // org.apache.hadoop.fs.aliyun.oss.AliyunOSSFileSystemStore.initialize(AliyunOSSFileSystemStore.java:111)
+  // org.apache.hadoop.fs.aliyun.oss.AliyunOSSFileSystem.initialize(AliyunOSSFileSystem.java:323)
+  // org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:3611)
+
+  implementation(libs.commons.lang)
```
Comment (jerryshao) [inline bundles/aliyun-bundle/build.gradle.kts:38]: Is it possible to shade this jar?

File: gradle/libs.versions.toml (hunk 1 above)
Comment (yuqi1129) [reply gradle/libs.versions.toml:166]: I have confirmed it and it seems there already exists the content in the LICENSE.bin


https://github.com/apache/gravitino/blob/93cdbc259b1b4bcd4bda2d1b34d011f19368d2dd/LICENSE.bin#L253

File: bundles/aliyun-bundle/build.gradle.kts (hunk 2 above)
Comment (yuqi1129) [reply bundles/aliyun-bundle/build.gradle.kts:38]: I think so, the package will be shaded in the jar `gravitino-aliyun-bundle`.

Comment (jerryshao) [reply bundles/aliyun-bundle/build.gradle.kts:38]: It is introduced by you, so you have to shade it.

Comment (yuqi1129) [reply bundles/aliyun-bundle/build.gradle.kts:37]: done.

File: LICENSE.bin [hunk 3]
```diff
@@ -282,6 +282,8 @@
    ApacheDS I18n
    ApacheDS Protocol Kerberos Codec
    Apache Hadoop
+   Apache Hadoop Aliyun connector
+   Apache Hadoop GCS connector
```
Comment (yuqi1129) [inline LICENSE.bin:286]: @jerryshao 
Please take a look if this change is acceptable.

**Review by jerryshao — APPROVED** [review]

**Pull request merged**